src/contracts/Chef.sol

```diff
@@ -55,6 +77,27 @@ abstract contract Chef is Ownable {
         return _rewardsActive;
     }
 
+    function getAccumulated() internal view virtual returns (uint256) {
+        uint256 blocksSince = block.number.sub(getLastRewardBlock());
+        return blocksSince.mul(getMetricPerBlock());
+    }
+
+    function getAcculatedWithmetricPrecision(uint accumulated) internal view virtual returns (uint) {
+        return accumulated.mul(ACC_METRIC_PRECISION);
+    }
+
+    function getTotalAllocationShares() public view returns (uint256) {
+        return _totalAllocShares;
+    }
+
+    function getLifetimeShareValue() public view returns (uint256) {
+        return _lifetimeShareValue;
+    }
+
+    function accumulatedMetricDividedByShares(uint256 accumulatedWithPrecision) public view returns (uint256) {
```

Comment:
is it possible to abstract away the ACC_METRIC_PRECISION from subclasses?  For example, could this function just take in accumulated metric and do the call to `getAcculatedWithmetricPrecision`?  It might be nice if the stakingChef/etc didn't have to know or care about this precision stuff

Reply:
I tried to make it modular setLifetimeShareValue - just calls 3 different functions within it so it's just a shortcut I guess
but you could call 
getAccumulated then whatever is returned from that call getAcculatedWithmetricPrecision and whatever is returned from that call accumulatedMetricDividedByShares and you can then take that value and add it to existing 
_lifetimeShareValue